src/main/java/org/fogbowcloud/saps/engine/core/archiver/storage/NfsPermanentStorage.java

```
@@ -1,20 +1,88 @@
+		String processingLocalDir = String.format(PROCESSING_DIR_PATTERN, sapsExports, taskId);
+
+		String nfsTaskDirPath = createTaskDir(task.getTaskId());
+		try {
+			copyDirToDir(inputdownloadingLocalDir, nfsTaskDirPath);
+			copyDirToDir(preprocessingLocalDir, nfsTaskDirPath);
+			copyDirToDir(processingLocalDir, nfsTaskDirPath);
+		} catch (IOException e) {
+			throw new PermanentStorageException("Error while copying local directories to nfs task dir [" + nfsTaskDirPath + "]", e);
+		}
+		return true;
 	}
 
 	@Override
 	public boolean delete(SapsImage task) throws PermanentStorageException {
-		// TODO Auto-generated method stub
-		return false;
+		String taskDirPath = String.format(TASK_DIR_PATTERN, nfsStoragePath, task.getTaskId());
+		File taskDir = new File(taskDirPath);
+		if (!taskDir.exists()) {
+			throw new PermanentStorageException("The task dir [" + taskDirPath + "] was not found on nfs storage directory [" + nfsStoragePath + "]");
+		}
+		try {
+			FileUtils.deleteDirectory(taskDir);
+		} catch (IOException e) {
+			throw new PermanentStorageException(e.getMessage(), e);
+		}
+		return true;
+	}
+
+	private void copyDirToDir(String src, String dest) throws IOException {
+		File srcDir = new File(src);
+		File destDir = new File(dest);
+		// The destination directory is created if it does not exist.
+		// If the destination directory did exist, then this method merges the source with the destination, with the source taking precedence.
+		LOGGER.debug("Copying [" + src + "] into [" + dest + "]");
+		FileUtils.copyDirectoryToDirectory(srcDir, destDir);
+	}
+
+	private String createTaskDir(String taskId) {
+		File storage_dir = new File(nfsStoragePath);
+		if (!storage_dir.exists()) {
+			throw new PermanentStorageException("The nfs storage directory [" + nfsStoragePath + "] was not found");
+		}
+		File nfsTaskDir = new File(String.format(TASK_DIR_PATTERN, nfsStoragePath, taskId));
+		try {
+			FileUtils.forceMkdir(nfsTaskDir);
```

I believe it would be better to let the method throw an exception of type `IOException` and add a catch to the code that calls it.

---

Done.